engfrosh_site/management/views.py

```diff
@@ -424,200 +419,6 @@ def manage_frosh_teams(request: HttpRequest) -> HttpResponse:
-                    return HttpResponseForbidden("Permission denied")
-
-                if "name" not in req_dict:
-                    return HttpResponseBadRequest("Missing new guild's name")
-
-                guild_res = DiscordGuild.create_new_guild(name=req_dict["name"])
-
-                return JsonResponse({"guild": {
-                    "name": guild_res.name,
-                    "id": guild_res.id
-                }})
-
-            case "get_one_use_invite":
-                if not request.user.has_perm("common_models.change_discordguild"):
-                    return HttpResponseForbidden("Permission denied")
-
-                if "guild_id" not in req_dict:
-                    return HttpResponseBadRequest("Missing guild id")
-
-                logger.debug(f"Invite requested with request body: {req_dict}")
-
-                try:
-                    guild = DiscordGuild.objects.get(id=int(req_dict["guild_id"]))
-                except DiscordGuild.DoesNotExist:
-                    return HttpResponseBadRequest("Guild with that id not available.")
-
-                invite: Invite = guild.create_invite(unique=True, max_uses=1)
-
-                return JsonResponse({
-                    "invite": {
-                        "full_url": invite.full_url,
-                        "code": invite.code
-                    }
-                })
-
-            case _:
-                return HttpResponseBadRequest("Invalid command.")
-
-    else:
-        return HttpResponseBadRequest("Bad http request method.")
-
-
-@permission_required("common_models.change_discordguild")
-def modify_discord_server(request: HttpRequest, slug: str) -> HttpResponse:
-
-    guild = DiscordGuild.objects.get(id=int(slug))
-
-    if request.method == "GET":
-
-        context = {
-            "guild": guild
-        }
-
-        return render(request, "modify_discord_server.html", context)
-
-    elif request.method == "POST":
-
-        if not request.user.has_perm("common_models.change_discordguild"):
-            return HttpResponseForbidden("Permission denied")
-
-        if request.content_type != "application/json":
-            return HttpResponseBadRequest("Invalid / missing content type.")
-
-        req_dict = json.loads(request.body)
-        if "command" not in req_dict:
-            return HttpResponseBadRequest("Bad request body, missing command.")
-
-        match req_dict["command"]:
-            case "add_admin":
-
-                role = guild.create_role("ADMIN", permissions=[Permissions.ADMINISTRATOR])
-
-                guild.add_role_to_member(discord_member_id=req_dict["discord_user_id"], discord_role=role)
-
-                return HttpResponse()
-
-            case _:
-                return HttpResponseBadRequest("Invalid command.")
-
-    else:
-        return HttpResponseNotAllowed(("GET",))
-
-
-def manage_discord_channel_groups(request: HttpRequest) -> HttpResponse:
```

Comment:
Same with the channels, but even more useful as it can lock many channels at a time.

Reply:
I have reverted this change but left a todo in order to rewrite it as I'm not 100% sure it works because a lot of the discord stuff seemed broken.